src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -67,31 +68,32 @@ internal sealed partial class SocketConnection : TransportConnection
 
             var pair = DuplexPipe.CreateConnectionPair(inputOptions, outputOptions);
 
-            // Set the transport and connection id
-            Transport = _originalTransport = pair.Transport;
+            _originalTransport = pair.Transport;
             Application = pair.Application;
 
+            Transport = new SocketDuplexPipe(this);
```

Comment:
Nit: 

```suggestion
            Transport = new SocketDuplexPipe(pair.Transport);
```

And remove `InnerTransport`.

Reply:
I kind prefer using properties rather than a private field, even if it's internal. This is purely a stylistic thing though.